apps/webapp/app/services/runs/getRun.server.ts

```diff
@@ -0,0 +1,70 @@
+import { PrismaClient, prisma } from "~/db.server";
+import { z } from "zod";
+
+const GetRunInputSchema = z.object({
+  runId: z.string(),
+  maxTasks: z.number().default(20),
+  taskDetails: z.boolean().default(false),
+  subTasks: z.boolean().default(false),
+  cursor: z.string().optional(),
+});
+
+export type GetRunInput = z.infer<typeof GetRunInputSchema>;
+
+export class GetRunService {
+  #prismaClient: PrismaClient;
+
+  constructor(prismaClient: PrismaClient = prisma) {
+    this.#prismaClient = prismaClient;
+  }
+
+  public async call(input: GetRunInput) {
+    const parsedInput = GetRunInputSchema.parse(input);
```

Comment:
Using a Zod schema here and parsing the input shouldn't be necessary. The "parsing" should all happen in the route and by the time `GetRunServce.call` is called, all the types should already be known.